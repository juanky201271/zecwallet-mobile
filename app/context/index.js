import { defaultAppContextLoaded, ContextAppLoaded, ContextAppLoadedProvider } from './contextAppLoaded';
import { defaultAppContextLoading, ContextAppLoading, ContextAppLoadingProvider } from './contextAppLoading';

export {
  defaultAppContextLoaded,
  ContextAppLoaded,
  ContextAppLoadedProvider,
  defaultAppContextLoading,
  ContextAppLoading,
  ContextAppLoadingProvider,
};
