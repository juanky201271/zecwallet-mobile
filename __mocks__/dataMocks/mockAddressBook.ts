import { AddressBookFileClass } from '../../app/AppState';

export const mockAddressBook: AddressBookFileClass[] = [
  {
    label: 'pepe',
    address: 'u1234567890_____________',
  },
  {
    label: 'lolo',
    address: 'u0987654321_____________',
  },
];
