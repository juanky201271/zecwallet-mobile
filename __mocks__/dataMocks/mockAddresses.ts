import { AddressClass, AddressKindEnum, ReceiverEnum } from '../../app/AppState';

export const mockAddresses: AddressClass[] = [
  {
    uaAddress: 'UA-12345678901234567890',
    address: 'UA-12345678901234567890',
    addressKind: AddressKindEnum.u,
    containsPending: false,
    receivers: ReceiverEnum.o + ReceiverEnum.z + ReceiverEnum.t,
  },
  {
    uaAddress: 'UA-12345678901234567890',
    address: 'sapling-12345678901234567890',
    addressKind: AddressKindEnum.z,
    containsPending: false,
    receivers: ReceiverEnum.z,
  },
  {
    uaAddress: 'UA-12345678901234567890',
    address: 'transparent-12345678901234567890',
    addressKind: AddressKindEnum.t,
    containsPending: false,
    receivers: ReceiverEnum.t,
  },
];
