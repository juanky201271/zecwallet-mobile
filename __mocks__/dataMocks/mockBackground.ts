import { BackgroundType } from '../../app/AppState';

export const mockBackground: BackgroundType = {
  batches: 0,
  message: '',
  date: 0,
  dateEnd: 0,
};
