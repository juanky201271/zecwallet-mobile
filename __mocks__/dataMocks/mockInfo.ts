import { ChainNameEnum, CurrencyNameEnum, InfoType } from '../../app/AppState';
import { serverUris } from '../../app/uris';

export const mockInfo: InfoType = {
  chainName: ChainNameEnum.mainChainName,
  serverUri: serverUris(() => {})[0].uri,
  latestBlock: 2000000,
  connections: 0,
  version: 'server_version',
  verificationProgress: 0,
  currencyName: CurrencyNameEnum.ZEC,
  solps: 0,
  zingolib: 'zingolib_version',
};
