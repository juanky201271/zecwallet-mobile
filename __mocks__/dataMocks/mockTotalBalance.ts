import { TotalBalanceClass } from '../../app/AppState';

export const mockTotalBalance: TotalBalanceClass = {
  transparentBal: 0.12345678,
  privateBal: 0.4,
  spendablePrivate: 0.2,
  orchardBal: 0.6,
  spendableOrchard: 0.3,
  total: 1.12345678,
};
