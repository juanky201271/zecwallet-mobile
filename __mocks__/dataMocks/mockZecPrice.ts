import { ZecPriceType } from '../../app/AppState';

export const mockZecPrice: ZecPriceType = {
  zecPrice: 33.33,
  date: 0,
};
