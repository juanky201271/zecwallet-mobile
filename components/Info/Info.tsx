/* eslint-disable react-native/no-inline-styles */
import React, { useContext } from 'react';
import { View, ScrollView, SafeAreaView } from 'react-native';
import { useTheme } from '@react-navigation/native';

import Button from '../Components/Button';
import DetailLine from '../Components/DetailLine';
import { ThemeType } from '../../app/types';
import { ContextAppLoaded } from '../../app/context';
import PriceFetcher from '../Components/PriceFetcher';
import Header from '../Header';
import CurrencyAmount from '../Components/CurrencyAmount';
import RegText from '../Components/RegText';
import moment from 'moment';
import 'moment/locale/es';
import 'moment/locale/pt';
import 'moment/locale/ru';
import { ButtonTypeEnum, ChainNameEnum, CurrencyEnum } from '../../app/AppState';

type InfoProps = {
  closeModal: () => void;
  setZecPrice: (p: number, d: number) => void;
};

const Info: React.FunctionComponent<InfoProps> = ({ closeModal, setZecPrice }) => {
  const context = useContext(ContextAppLoaded);
  const { info, translate, currency, zecPrice, privacy, language } = context;
  const { colors } = useTheme() as unknown as ThemeType;
  moment.locale(language);

  return (
    <SafeAreaView
      style={{
        display: 'flex',
        justifyContent: 'flex-start',
        alignItems: 'stretch',
        height: '100%',
        backgroundColor: colors.background,
      }}>
      <Header
        title={translate('info.title') as string}
        noBalance={true}
        noSyncingStatus={true}
        noDrawMenu={true}
        noPrivacy={true}
      />

      <ScrollView
        style={{ maxHeight: '85%' }}
        contentContainerStyle={{
          flexDirection: 'column',
          alignItems: 'stretch',
          justifyContent: 'flex-start',
        }}>
        <View style={{ display: 'flex', margin: 20, marginBottom: 30 }}>
          <DetailLine
            label={translate('info.version') as string}
            value={translate('zingo') + ' ' + translate('version')}
          />
          <DetailLine
            label={translate('info.serverversion') as string}
            value={info.version ? info.version : (translate('loading') as string)}
          />
          <DetailLine
            label={translate('info.lightwalletd') as string}
            value={info.serverUri ? info.serverUri : (translate('loading') as string)}
          />
          <DetailLine
            label={translate('info.network') as string}
            value={
              !info.chainName
                ? (translate('loading') as string)
                : info.chainName === ChainNameEnum.mainChainName
                ? 'Mainnet'
                : info.chainName === ChainNameEnum.testChainName
                ? 'Testnet'
                : info.chainName === ChainNameEnum.regtestChainName
                ? 'Regtest'
                : (translate('info.unknown') as string) + ' (' + info.chainName + ')'
            }
          />
          <DetailLine
            label={translate('info.serverblock') as string}
            value={info.latestBlock ? info.latestBlock.toString() : (translate('loading') as string)}
          />
          {currency === CurrencyEnum.USDCurrency && (
            <View style={{ flexDirection: 'row', alignItems: 'flex-end' }}>
              <DetailLine label={translate('info.zecprice') as string}>
                {zecPrice.zecPrice === -1 && (
                  <RegText color={colors.text}>{translate('info.errorgemini') as string}</RegText>
                )}
                {zecPrice.zecPrice === -2 && (
                  <RegText color={colors.text}>{translate('info.errorrpcmodule') as string}</RegText>
                )}
                <CurrencyAmount price={zecPrice.zecPrice} amtZec={1} currency={currency} privacy={privacy} />
              </DetailLine>
              <View style={{ marginLeft: 5 }}>
                <PriceFetcher setZecPrice={setZecPrice} />
              </View>
            </View>
          )}
        </View>
      </ScrollView>

      <View
        style={{
          flexGrow: 1,
          flexDirection: 'row',
          justifyContent: 'center',
          alignItems: 'center',
          marginVertical: 5,
        }}>
        <Button type={ButtonTypeEnum.Secondary} title={translate('close') as string} onPress={closeModal} />
      </View>
    </SafeAreaView>
  );
};

export default Info;
