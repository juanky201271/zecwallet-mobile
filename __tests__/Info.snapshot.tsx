/**
 * @format
 */

import 'react-native';
import React from 'react';

import { render } from '@testing-library/react-native';
import Info from '../components/Info';
import { defaultAppContextLoaded, ContextAppLoadedProvider } from '../app/context';
import { mockInfo } from '../__mocks__/dataMocks/mockInfo';
import { mockTranslate } from '../__mocks__/dataMocks/mockTranslate';
import { mockTotalBalance } from '../__mocks__/dataMocks/mockTotalBalance';
import { mockZecPrice } from '../__mocks__/dataMocks/mockZecPrice';

jest.useFakeTimers();
jest.mock('@fortawesome/react-native-fontawesome', () => ({
  FontAwesomeIcon: '',
}));
jest.mock('react-native-localize', () => ({
  getNumberFormatSettings: () => {
    return {
      decimalSeparator: '.', // us
      groupingSeparator: ',', // us
    };
  },
}));
jest.mock('react-native/Libraries/Animated/NativeAnimatedHelper');
jest.mock('@react-native-community/netinfo', () => {
  const RN = jest.requireActual('react-native');

  RN.NativeModules.RNCNetInfo = {
    execute: jest.fn(() => '{}'),
  };

  return RN;
});
jest.mock('react-native', () => {
  const RN = jest.requireActual('react-native');

  RN.NativeModules.RPCModule = {
    execute: jest.fn(() => '{}'),
  };

  return RN;
});

// test suite
describe('Component Info - test', () => {
  //snapshot test
  test('Info - snapshot', () => {
    const state = defaultAppContextLoaded;
    state.info = mockInfo;
    state.zecPrice = mockZecPrice;
    state.translate = mockTranslate;
    state.totalBalance = mockTotalBalance;
    const onClose = jest.fn();
    const onSet = jest.fn();
    const info: any = render(
      <ContextAppLoadedProvider value={state}>
        <Info closeModal={onClose} setZecPrice={onSet} />
      </ContextAppLoadedProvider>,
    );
    expect(info.toJSON()).toMatchSnapshot();
  });
});
