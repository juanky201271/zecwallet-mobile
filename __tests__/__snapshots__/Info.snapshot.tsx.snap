// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Info - test Info - snapshot 1`] = `
<RCTSafeAreaView
  style={
    {
      "alignItems": "stretch",
      "backgroundColor": "rgb(242, 242, 242)",
      "display": "flex",
      "height": "100%",
      "justifyContent": "flex-start",
    }
  }
>
  <View>
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "rgb(255, 255, 255)",
          "display": "flex",
          "minHeight": 25,
          "paddingBottom": 0,
          "paddingTop": 10,
          "zIndex": -1,
        }
      }
      testID="header"
    >
      <View
        style={
          {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
            "flexWrap": "wrap",
            "justifyContent": "center",
            "marginHorizontal": 5,
            "marginTop": 12,
          }
        }
      />
    </View>
    <View
      style={
        {
          "left": 0,
          "padding": 11.5,
          "position": "absolute",
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "height": 45,
          }
        }
      />
    </View>
    <View
      style={
        {
          "padding": 15,
          "position": "absolute",
          "right": 0,
        }
      }
    >
      <Image
        source={
          {
            "testUri": "../../../assets/img/logobig-zingo.png",
          }
        }
        style={
          {
            "borderRadius": 10,
            "height": 38,
            "resizeMode": "contain",
            "width": 38,
          }
        }
      />
    </View>
  </View>
  <View>
    <View
      style={
        {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
          "flexWrap": "wrap",
          "justifyContent": "space-between",
          "marginVertical": 5,
          "width": "100%",
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      >
        <
          color="rgb(0, 122, 255)"
          icon={
            {
              "icon": [
                320,
                512,
                [
                  9001,
                ],
                "f053",
                "M9.4 233.4c-12.5 12.5-12.5 32.8 0 45.3l192 192c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L77.3 256 246.6 86.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0l-192 192z",
              ],
              "iconName": "chevron-left",
              "prefix": "fas",
            }
          }
          size={30}
          style={
            {
              "marginHorizontal": 10,
            }
          }
        />
      </View>
      <Text
        style={
          {
            "color": "rgb(28, 28, 30)",
            "fontSize": 18,
            "fontWeight": "600",
            "opacity": 1,
            "paddingHorizontal": 5,
          }
        }
      >
        text translated
      </Text>
      <View
        style={
          {
            "height": 30,
            "marginHorizontal": 10,
            "width": 30,
          }
        }
      />
    </View>
    <View
      style={
        {
          "backgroundColor": "rgb(0, 122, 255)",
          "height": 1,
          "width": "100%",
        }
      }
    />
  </View>
  <RCTScrollView
    contentContainerStyle={
      {
        "alignItems": "stretch",
        "flexDirection": "column",
        "justifyContent": "flex-start",
      }
    }
    style={
      {
        "maxHeight": "90%",
      }
    }
  >
    <View>
      <View
        style={
          {
            "display": "flex",
            "margin": 20,
            "marginBottom": 30,
          }
        }
      >
        <View
          style={
            {
              "display": "flex",
              "marginTop": 20,
            }
          }
        >
          <Text
            style={
              {
                "color": "rgb(28, 28, 30)",
                "opacity": 0.65,
              }
            }
          >
            text translated
          </Text>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "opacity": 1,
              }
            }
          >
            <Text
              style={
                {
                  "color": "rgb(28, 28, 30)",
                  "fontSize": 18,
                  "fontWeight": "600",
                  "opacity": 1,
                }
              }
            >
              text translated text translated
            </Text>
          </View>
        </View>
        <View
          style={
            {
              "display": "flex",
              "marginTop": 20,
            }
          }
        >
          <Text
            style={
              {
                "color": "rgb(28, 28, 30)",
                "opacity": 0.65,
              }
            }
          >
            text translated
          </Text>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "opacity": 1,
              }
            }
          >
            <Text
              style={
                {
                  "color": "rgb(28, 28, 30)",
                  "fontSize": 18,
                  "fontWeight": "600",
                  "opacity": 1,
                }
              }
            >
              server_version
            </Text>
          </View>
        </View>
        <View
          style={
            {
              "display": "flex",
              "marginTop": 20,
            }
          }
        >
          <Text
            style={
              {
                "color": "rgb(28, 28, 30)",
                "opacity": 0.65,
              }
            }
          >
            text translated
          </Text>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "opacity": 1,
              }
            }
          >
            <Text
              style={
                {
                  "color": "rgb(28, 28, 30)",
                  "fontSize": 18,
                  "fontWeight": "600",
                  "opacity": 1,
                }
              }
            >
              https://zec.rocks:443
            </Text>
          </View>
        </View>
        <View
          style={
            {
              "display": "flex",
              "marginTop": 20,
            }
          }
        >
          <Text
            style={
              {
                "color": "rgb(28, 28, 30)",
                "opacity": 0.65,
              }
            }
          >
            text translated
          </Text>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "opacity": 1,
              }
            }
          >
            <Text
              style={
                {
                  "color": "rgb(28, 28, 30)",
                  "fontSize": 18,
                  "fontWeight": "600",
                  "opacity": 1,
                }
              }
            >
              Mainnet
            </Text>
          </View>
        </View>
        <View
          style={
            {
              "display": "flex",
              "marginTop": 20,
            }
          }
        >
          <Text
            style={
              {
                "color": "rgb(28, 28, 30)",
                "opacity": 0.65,
              }
            }
          >
            text translated
          </Text>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "opacity": 1,
              }
            }
          >
            <Text
              style={
                {
                  "color": "rgb(28, 28, 30)",
                  "fontSize": 18,
                  "fontWeight": "600",
                  "opacity": 1,
                }
              }
            >
              2000000
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;
